### PR TITLE
Debug markers

### LINKED ISSUE #43: Search Location feature to Create Marker form
Adapt the search box that is attached to the map to allow the user to search for where they want to place the marker (inside the form)

### LINKED ISSUE #50: Markers servlet always returns empty JSON array
https://github.com/googleinterns/step125-2020/blob/master/maptivist/src/main/java/com/google/sps/servlets/MarkerServlet.java#L47-L53

An empty `markers` array is created, but then the markers are stored in a `results` array.

Then response then converts the still-empty `markers` into JSON and returns it. It should be converting the `results` array into JSON instead. In fact, we should just delete the `markers` array altogether.

### PR DESCRIPTION
**Pull request by TBarrett553**
Markers are now displayed based on map boundary and are fetched from the Marker Servlet.

### CODE REVIEW

File: maptivist/src/main/webapp/index.html [hunk 1]
```diff
@@ -8,8 +8,8 @@
     <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-auth.js"></script>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
     <script src="script.js"></script>
```
Comment (flying-bus) [inline maptivist/src/main/webapp/index.html:11]: Is this line now redundant with Line 12?

File: maptivist/src/main/webapp/script.js [hunk 2]
```diff
@@ -171,7 +99,8 @@ function searchBox(){
 function loadMarkersByBoundary() {
   fetch("/marker").then(response => response.json()).then((markers) => {
     var markersToDisplay = markers.filter(inBoundary);
-    for (let marker in markersToDisplay) {
+    var marker;
+    for (marker of markersToDisplay) {
```
Comment (flying-bus) [inline maptivist/src/main/webapp/script.js]: Does it work if you do:

`for (const marker of markersToDisplay) {`

as a single line?

File: maptivist/src/main/webapp/script.js [hunk 3]
```diff
@@ -181,79 +110,74 @@ function inBoundary(marker){
...
   return (marker.latitude <= bounds.north && 
          marker.latitude >= bounds.south &&
-         marker.longitude <= bounds.west &&
-         marker.longitude >= bounds.east);
+         marker.longitude >= bounds.west &&
+         marker.longitude <= bounds.east);
 }
 
 /** Adds a new Marker based on id array
  */
```
Comment (flying-bus) [inline maptivist/src/main/webapp/script.js]: Can you update this comment?

The input value is a Marker in json format, right?

File: maptivist/src/main/webapp/script.js [hunk 4]
```diff
@@ -181,79 +110,74 @@ function inBoundary(marker){
...
 }
 
-
-function drawInfowindow(id, title, description, links, categories, position) {
-  //Set info window content from Marker class
-
-  var location = position.toString();
-  
+/**
+* Adds an infowindow based on the marker creation form
```
Comment (flying-bus) [inline maptivist/src/main/webapp/script.js]: Add what "markerObj" is in the documentation here. (a json object).

**Review by flying-bus — APPROVED** [review]
Just a few small suggestions. Looks good!

**Review by ayomide321 — APPROVED** [review]

File: maptivist/src/main/webapp/index.html (hunk 1 above)
Comment (TBarrett553) [reply maptivist/src/main/webapp/index.html:11]: Yes it is I think I merged it incorrectly

File: maptivist/src/main/webapp/script.js (hunk 3 above)
Comment (TBarrett553) [reply maptivist/src/main/webapp/script.js]: Yes that's correct let me make that change

File: maptivist/src/main/webapp/script.js (hunk 4 above)
Comment (TBarrett553) [reply maptivist/src/main/webapp/script.js]: It's a single marker object that is a part of the JSON array

**Pull request merged**